### ISSUE TITLE
pkg_zip removes all directories when packing

### ISSUE THREAD

**Issue by spion-h4** [bazelbuild/rules_pkg#170]
pkg_zip uses an invocation of `dest_path` that triggers a buggy edge case (when strip_path is None)

https://github.com/bazelbuild/rules_pkg/blob/06e57ddaa6220ade969e710832d84ca1a377b522/pkg/pkg.bzl#L388-L389

The faulty logic for the None case in dest_path causes it to strip the entire (package-relative) directory of a file:

https://github.com/bazelbuild/rules_pkg/blob/7a991dea418ab17c7e86f0a7b5e7d4a87ef4304b/pkg/path.bzl#L42-L49

Running `bazel` with the `-s` switch demonstrates that this is indeed what gets passed to the python tool:

```
bazel-out/host/bin/external/rules_pkg/build_zip 
  -o bazel-out/darwin-fastbuild/bin/packages/aaa/preTokenLambda/preTokenLambda_zip.zip 
  -d / 
  -t 315532800 
  'bazel-out/darwin-fastbuild/bin/packages/aaa/preTokenLambda/preTokenLambda_npm/README.md=/README.md' 
  'bazel-out/darwin-fastbuild/bin/packages/aaa/preTokenLambda/preTokenLambda_npm/_BUILD=/_BUILD' 
  'bazel-out/darwin-fastbuild/bin/packages/aaa/preTokenLambda/preTokenLambda_npm/index.d.ts=/index.d.ts' 
  'bazel-out/darwin-fastbuild/bin/packages/aaa/preTokenLambda/preTokenLambda_npm/index.js=/index.js' 
  'bazel-out/darwin-fastbuild/bin/packages/aaa/preTokenLambda/preTokenLambda_npm/package.json=/package.json' 
  'bazel-out/darwin-fastbuild/bin/packages/aaa/preTokenLambda/preTokenLambda_npm/src/preTokenLambda.d.ts=/preTokenLambda.d.ts'

**Comment by spion-h4** [comment]
I'm assuming we want `strip_prefix = False` which will simply return the `safe_short_path` here?

**Comment by aiuto** [comment]
Fixed in #221.  I think.
